mlibrary/pride
# Pride.Core.FacetSearch

# Overview
`Pride.Core.FacetSearch` has been simplified, and tests have been written for `uid`, `getData`, `getResults`, `getMute`, `setMute`, `observables`, `resultsObservers`, `setDataObservers`, `runDataObservers`, and `clearAllObservers`.

## `notify`
`notify` was never used and has been removed.

## Testing
1. Build the repository (`npm run build`).
2. Run the tests to make sure they pass (`npm run test`).
   * Break the new/updated unit tests to make sure they're working properly.
3. Navigate to your local [Search repository](https://github.com/mlibrary/search) and apply the newly generated files:
   1. Open `./package.json` and add `#Pride-Core-FacetSearch` at the end of the `pride` dependency URL:
      ```bash
      "pride": "git+https://github.com/mlibrary/pride.git#Pride-Core-FacetSearch"
      ``` 
   2. Do a clean install of Search:
      ```bash
      rm -rf node_modules && rm package-lock.json && npm install
      ``` 
   3. Start Search (`npm start`) and look around [the site](http://localhost:3000/everything). Check to see if there are any console errors, and everything still works as expected.
- Make sure the PR is consistent in these browsers:
